### PR TITLE
[c++] Install span header

### PR DESCRIPTION
**Pull request by nguyenv**
**Issue and/or context:**

[This error](https://github.com/single-cell-data/TileDB-SOMA/pull/1302#issuecomment-1527799151) is occuring because installing span was accidentally removed from `CMakeLists.txt` in #1301.

**Changes:**

Install the span header located in the `externals` directory.

### CODE REVIEW

File: libtiledbsoma/src/CMakeLists.txt [hunk 1]
```diff
@@ -173,6 +173,7 @@ install(DIRECTORY
   ${CMAKE_CURRENT_SOURCE_DIR}/tiledbsoma
   ${CMAKE_CURRENT_SOURCE_DIR}/soma
   ${CMAKE_CURRENT_SOURCE_DIR}/utils
+  ${CMAKE_CURRENT_SOURCE_DIR}/../external/include/span
```
Comment (eddelbuettel) [inline libtiledbsoma/src/CMakeLists.txt:176]: Maybe external/ should be inside src/?  Or maybe CMakeLiists.txt should be one dir up?   Having a variable 'source dir' and then appending '..' feels a bit weird.

**Review by eddelbuettel — APPROVED** [review]
All for unblocking the fail but please see my one comment.

**Comment by johnkerl** [comment]
@nguyenv I need to be AFK for a bit (half hour or a smidge more) and I will try this on my #1302 scenario ASAP

Thank you!! :)

**Review by johnkerl — APPROVED** [review]
LGTM! :) Thanks!!!

With reference to https://github.com/single-cell-data/TileDB-SOMA/pull/1301#issuecomment-1527776132, here is with this PR:

```
$ scripts/bld --prefix=./testing

$ find ./testing -follow -type f | sort
testing/bin/tdbsoma
testing/include/cli/arrow_adapter.h
testing/include/cli/carrow.h
testing/include/cli/cli.cc
testing/include/soma/managed_query.cc
testing/include/soma/managed_query.h
testing/include/soma/soma_array.cc
testing/include/soma/soma_array.h
testing/include/span/span.hpp <------ YAY!
testing/include/tiledbsoma/logger_public.h
testing/include/tiledbsoma/tiledbsoma
testing/include/tiledbsoma/tiledbsoma_export.h
testing/include/utils/array_buffers.h
testing/include/utils/column_buffer.cc
testing/include/utils/column_buffer.h
testing/include/utils/common.h
testing/include/utils/logger.cc
testing/include/utils/logger.h
testing/include/utils/stats.cc
testing/include/utils/stats.h
testing/include/utils/util.cc
testing/include/utils/util.h
testing/include/utils/version.cc
testing/include/utils/version.h
testing/lib/cmake/tiledbsoma/TileDBSomaConfig.cmake
testing/lib/cmake/tiledbsoma/TileDBSomaTargets-release.cmake
testing/lib/cmake/tiledbsoma/TileDBSomaTargets.cmake
testing/lib/libtiledbsoma.dylib
testing/lib/pkgconfig/tiledbsoma.pc
```

**Pull request merged**